ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -81,6 +105,74 @@ const ApproveStaticSimulation = () => {
   );
 };
 
+const SpendingCap = () => {
+  const t = useI18nContext();
+
+  const transactionMeta = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const { receivedToken } = useReceivedToken();
+
+  const { tokenAmount, formattedTokenNum, value, pending } =
+    useApproveTokenSimulation(transactionMeta);
+
+  if (pending) {
+    return <Container isLoading />;
+  }
+
+  if (!receivedToken) {
+    return null;
+  }
+
+  return (
+    <ConfirmInfoSection>
+      <ConfirmInfoRow label={t('accountBalance')}>
+        <ConfirmInfoRowText
+          text={`${receivedToken.string} ${receivedToken.symbol}`}
+        />
+      </ConfirmInfoRow>
+
+      {value ? (
```

Comment:
Minor, rather than nesting the majority of the block, could we have a component such as `SpendingCapValue` that can early return?